src/App.tsx

```diff
@@ -1,56 +1,41 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { PeopleSearchBar } from './components/PeopleSearchBar';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+
+  const filteredPeople = useMemo(() => {
+    return [...peopleFromServer]
+      .filter((person) => person.name.toLowerCase()
+        .includes(appliedQuery.toLowerCase()));
+  }, [peopleFromServer, appliedQuery]);
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {selectedUser ? (
+        <h1 className="title">
+          {`${selectedUser.name} (${selectedUser.born} = ${selectedUser.died})`}
+        </h1>
+      ) : (
+        <h1 className="title">No selected person</h1>
+      )}
 
       <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
+        <PeopleSearchBar
+          onType={setQuery}
+          query={query}
+          setAppliedQuery={setAppliedQuery}
+          delay={1000}
+        />
+        {appliedQuery && (
```

Comment:
show the list of all people when input is focused but empty
![Screenshot from 2023-07-30 17-43-01](https://github.com/mate-academy/react_autocomplete/assets/33029864/1fcd4e75-62c2-4f4a-8bc7-7455955d9830)